lib/screens/v2/candidate_generator/dup/departures.ex

```diff
@@ -0,0 +1,251 @@
+      if secondary_sections == [] do
+        primary_sections
+      else
+        secondary_sections
+      end
+
+    secondary_departures_instances =
+      secondary_sections
+      |> get_sections_data(fetch_section_departures_fn, fetch_alerts_fn, now)
+      |> sections_data_to_departure_instances(
+        config,
+        [:main_content_two],
+        now
+      )
+
+    primary_departures_instances ++ secondary_departures_instances
+  end
+
+  defp sections_data_to_departure_instances(sections_data, config, slot_ids, now) do
+    if Enum.any?(sections_data, &(&1 == :error)) do
+      %DeparturesNoData{screen: config, show_alternatives?: true}
+    else
+      sections =
+        Enum.map(sections_data, fn %{
+                                     departures: departures,
+                                     pill: pill,
+                                     alert: alert,
+                                     headway: headway,
+                                     stop_ids: stop_ids
+                                   } ->
+          visible_departures =
+            if length(sections_data) > 1 do
+              Enum.take(departures, 2)
+            else
+              Enum.take(departures, 4)
+            end
+
+          case get_headway_mode(stop_ids, headway, alert, now) do
+            {:active, time_range, headsign} ->
+              %{type: :headway_section, pill: pill, time_range: time_range, headsign: headsign}
+
+            :inactive ->
+              %{type: :normal_section, rows: visible_departures}
+          end
+        end)
+
+      Enum.map(slot_ids, fn slot_id ->
+        %DeparturesWidget{
+          screen: config,
+          section_data: sections,
+          slot_names: [slot_id]
+        }
+      end)
+    end
+  end
+
+  defp get_sections_data(sections, fetch_section_departures_fn, fetch_alerts_fn, now) do
+    sections
+    |> Task.async_stream(fn %Section{
+                              query: %Query{
+                                params: %Params{stop_ids: stop_ids} = params
+                              },
+                              headway: headway
+                            } = section ->
+      section_departures =
+        case fetch_section_departures_fn.(section) do
+          {:ok, section_departures} -> section_departures
+          _ -> []
+        end
+
+      section_alert = get_section_alert(params, fetch_alerts_fn, now)
+
+      section_route = get_section_route_from_alert(stop_ids, section_alert)
+
+      %{
+        departures: section_departures,
+        pill: section_route,
+        alert: section_alert,
+        headway: headway,
+        stop_ids: stop_ids
+      }
+    end)
+    |> Enum.map(fn {:ok, data} -> data end)
+  end
+
+  # Alert will only have a value for sections that are configured for a station's ID
+  defp get_section_route_from_alert(["place-" <> _ = stop_id], %Alert{
+         informed_entities: informed_entities
+       }) do
+    informed_entities
```

Comment:
Check out the alerts currently in dev-green. I added a print statement for `informed_entities` here, and somehow it is misshapen: `[%{direction_id: nil, facility: nil, route: "Red", route_type: 1, stop: nil}]`. If the `Enum.find_value` below this fails and returns nil, then the `downcase()` breaks.

Reply:
Good catch. I changed the default to be an empty string instead of `nil`. I also moved this so we don't call it unless we need it. In this case, that alert would never cause headway mode. Getting the pill color for an alert that will never show as headway mode is unnecessary.